policies/codingstyle.txt

```diff
@@ -289,12 +289,78 @@ options.
 
 In source files, separate functions with one blank line.
 
+                Chapter 6.1: Function Arguments
+
+Most API functions in OpenSSL revolve around a specific type (e.g. SSL_CTX,
+EVP_PKEY, X509 etc). When adding new API functions relevant to a given type, the
+following argument ordering should be used:
+
+    rettype TYPE_get_something(TYPE *,[args])
+    rettype TYPE_set_something(TYPE *,[args])
+    rettype TYPE_do_something(TYPE *,[args])
+    TYPE *TYPE_new([args])
+
+Some more complex complex APIs have a TYPE and TYPE_CTX pair, where TYPE usually
+represents the set of function pointers to the implementation, while TYPE_CTX is
+the collection of attributes to perform the intended operation.
+For that case, the pattern should also include:
+
+    TYPE_CTX *TYPE_CTX_new([args])
+    rettype TYPE_CTX_get_something(TYPE_CTX *,[args])
+    rettype TYPE_CTX_set_something(TYPE_CTX *,[args])
+    rettype TYPE_do_something(TYPE_CTX *,[args])
+
+Where present the TYPE (or TYPE_CTX) argument should always be first as shown
+in the examples above. In constructors or object factory type functions (such as
+a "fetch" function), the key argument that identifies the thing be constructed
+should be placed first (if such an argument exists). For example in a "fetch"
+function the name of the algorithm being fetched would be first.
+
+The remaining argument list should be put into order of importance. Deciding the
+relative importance of arguments is necessarily somewhat subjective, but the
+following guidelines should be considered:
+
+1) Any additional "context" arguments should be first. If multiple contexts are
```

Comment:
Why the double quotes around context?

Reply:
From https://en.wikipedia.org/wiki/Quotation_marks_in_English

"They are also used to indicate that the meaning of the word or phrase they surround should be taken to be different from (or, at least, a modification of) that typically associated with it "

So here it is modifying the normal english meaning of context to specifically refer to "TYPE_CTX" context arguments.